lib/src/client/connection.dart

```diff
@@ -39,176 +39,26 @@ enum ConnectionState {
   shutdown
 }
 
-class ClientConnection {
-  final ChannelOptions options;
-  final Future<Transport> Function() _connectTransport;
+abstract class ClientConnection {
```

Comment:
I'm starting to think this abstraction could disappear too... as there is no need to expose it from `ClientChannel`

Reply:
Yes - I think at least from the outside.